pkg/cmd/cli/cmd/rsync/copyrsync.go

```diff
@@ -24,13 +26,23 @@ type rsyncStrategy struct {
 	RemoteExecutor executor
 }
 
+var rshExcludeFlags = map[string]struct{}{
+	"delete":   {},
+	"strategy": {},
+	"quiet":    {},
+}
+
 func newRsyncStrategy(f *clientcmd.Factory, c *cobra.Command, o *RsyncOptions) (copyStrategy, error) {
 	// Determine the rsh command to pass to the local rsync command
 	rsh := siblingCommand(c, "rsh")
-	rshCmd := []string{rsh, "-n", o.Namespace}
-	if len(o.ContainerName) > 0 {
-		rshCmd = append(rshCmd, "-c", o.ContainerName)
-	}
+	rshCmd := []string{rsh}
+	// Append all original flags to rsh command
+	c.Flags().Visit(func(flag *pflag.Flag) {
+		if _, exclude := rshExcludeFlags[flag.Name]; exclude {
+			return
+		}
+		rshCmd = append(rshCmd, fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()))
```

Comment:
> this seems really odd to me... a blacklist we have to keep updated whenever we add options to rsync?

Yes, unless we have a list somewhere of all the global flags. I can add a comment to the code that adds the flags in the main command to update this list if adding a new one.



> will all fields be string-convertible?

Yes, because they implement the pflag.Value interface